craft_parts/features.py

```diff
@@ -36,6 +37,24 @@ class Features(metaclass=Singleton):
     enable_overlay: bool = False
     enable_partitions: bool = False
 
+    def __init__(
+        self, *, enable_overlay: bool = False, enable_partitions: bool = False
+    ) -> None:
+        """Initialize craft-parts features.
+
+        :param enable_overlay: Enables the overlay step.
+        :param enable_partitions: Enables the usage of partitions.
+
+        :raises FeatureError: If mutually exclusive features are enabled.
+        """
+        if enable_overlay and enable_partitions:
```

Comment:
Since this is a dataclass, this might be better to do in `__post_init__`. As a side benefit we won't have to touch `__init__` when we add other features.

Reply:
Ack, great idea!